genjava/service_spring_controllers.go

```diff
@@ -50,23 +50,23 @@ func generateController(version *spec.Version, api *spec.Api, apiPackage Module,
 	w.Line(`  private ObjectMapper objectMapper;`)
 	for _, operation := range api.Operations {
 		w.EmptyLine()
-		generateMethod(w.Indented(), version, api, operation)
+		generateMethod(w.Indented(), api, operation)
 	}
 	w.Line(`}`)
 
 	files = append(files, gen.TextFile{
-		Path:    apiPackage.GetPath(fmt.Sprintf("%s.java", controllerName(api))),
+		Path:    apiPackage.GetPath(fmt.Sprintf("%s.java", className)),
 		Content: w.String(),
 	})
 
 	return files
 }
 
-func generateMethod(w *gen.Writer, version *spec.Version, api *spec.Api, operation spec.NamedOperation) {
+func generateMethod(w *gen.Writer, api *spec.Api, operation spec.NamedOperation) {
```

Comment:
1. Pass operation as a reference instead of value.
2. operation.Api is a reference to the API - no need to pass api separately.